src/index.js

```diff
@@ -240,7 +337,7 @@ class MiniCssExtractPlugin {
       const { mainTemplate } = compilation;
 
       mainTemplate.hooks.localVars.tap(pluginName, (source, chunk) => {
```

Comment:
All this runtime code should use a RuntimeModule in webpack 5.

Something like `CssChunkLoadingRuntimeModule`.

A runtime module can be added like that:

``` js
			compilation.hooks.runtimeRequirementInTree
				.for(RuntimeGlobals.ensureChunkHandlers)
				.tap(pluginName, (chunk, set) => {
					compilation.addRuntimeModule(chunk, new CssChunkLoadingRuntimeModule());
				});
```

Take a look at `ReadFileChunkLoadingRuntimeModule` in webpack for an example how to do that.

Technically you can put the HMR logic there too, but we can leave this for another PR.